### PR TITLE
[Feature] html artifact view

### PR DESCRIPTION
**Pull request by sim-san**
## What changes are proposed in this pull request?
Added functionality to display artifact previews for html sites inside the MLFlow server (e.g. Poltly plots).

This is an example of how it looks:
FR_Mlflow
 
## How is this patch tested?
Manual test. Only with a simple html page and a ploty plot.
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
Added support to display artifact previews for html sites.

### What component(s) does this PR affect?
 
- [x] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [x] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by mateiz** [comment]
Hi Simon,

One concern about this is whether the iframe will allow potentially untrusted code to run from the same origin as the MLflow UI. This would allow that code to take actions through MLflow's REST API, e.g. log runs or delete all your runs or stuff like that. Does your use case require JavaScript within the iframe? We should try to sandbox it with the iframe sandbox attribute: https://www.w3schools.com/tags/att_iframe_sandbox.asp.

**Comment by sim-san** [comment]
> Does your use case require JavaScript within the iframe?

Yes, it is necessary to be able to execute Javascript for example to render a plot of Plotly. Ich will check the sandbox option of the iframe.

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  render() {
+    if (this.state.loading) {
+      return (
+        <div>
+          Loading...
+        </div>
+      );
+    }
+    if (this.state.error) {
+      return (
```
Comment (smurching) [inline mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js]: Could we also log the error to help users debug/give us diagnostic information in the event of failures? e.g.
```logger.error("Unable to load HTML artifact, got error " + this.state.error")```?

**Review by smurching — COMMENTED** [review]
@sim-san this looks pretty reasonable with the `sandbox` attribute update - I'll confirm that we're good on that front with some security-minded folks, otherwise this looks pretty good to me!

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js (hunk 1 above)
Comment (sim-san) [reply mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js]: Do you expect me to print the error on the web console?
`console.error("Unable to load HTML artifact, got error " + this.state.error);`

Comment (smurching) [reply mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js]: Yep, let's add that line before we `return` here, e.g.

```suggestion
      console.error("Unable to load HTML artifact, got error " + this.state.error);
      return (
```

assuming that `this.state.error` is in fact a string (would be great if you could confirm this) - thanks!

Comment (sim-san) [reply mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js]: I have copy the fetchArtifacts() function from the ShowArtifactTextView.js file. I currently can't see where an error is being set.

Comment (sim-san) [reply mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js]: I add a catch() method to set the error in a error case

**Comment by smurching** [comment]
BTW, one last comment - it'd be great to add some component rendering tests analogous to those added in https://github.com/mlflow/mlflow/pull/1556/files#diff-b0bb5567cc37de50eeeffde66f9653b3, e.g. add a test verifying we can render a HTML artifact containing a div containing some specific text, etc.

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js [hunk 2]
```diff
@@ -83,7 +82,7 @@ class ShowArtifactHtmlView extends Component {
         this.setState({ html: event.target.result, loading: false });
       };
       fileReader.readAsText(blob);
-    });
+    }).catch(error => this.setState({ error: error, loading: false }));;
```
Comment (sim-san) [inline mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js]: This should also be added to the ShowArtifactTextView.js. Should I open a new pull request for this ?

**Comment by sim-san** [comment]
> 
> 
> BTW, one last comment - it'd be great to add some component rendering tests analogous to those added in https://github.com/mlflow/mlflow/pull/1556/files#diff-b0bb5567cc37de50eeeffde66f9653b3, e.g. add a test verifying we can render a HTML artifact containing a div containing some specific text, etc.

How can I run this rendering tests ?

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js (hunk 2 above)
Comment (smurching) [reply mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js]: Yep, a separate PR for this in ShowArtifactTextView.js makes sense - ideally in that separate Pr, we could pull `fetchArtifacts` out into a utility so we don't need to copy the implementation across components :)

**Comment by smurching** [comment]
@sim-san thanks for working on the test - I tried to debug it some & found that the particular code pattern here (using the `fetch` and `FileReader` APIs) is surprisingly hard to test within our JS testing framework due to unexpected behavior when e.g. trying to pass a dummy blob to the `FileReader`. The test I ended up with effectively just verifies that the iframe uses `this.state.html` as its source, so decided not to include it. I'll ask around about this tomorrow, but I don't think we should block this PR on those challenges, so will go ahead & merge the PR!

**Pull request merged**